staging/src/k8s.io/component-base/config/serializer/encode.go

```diff
@@ -0,0 +1,93 @@
+package serializer
```

Comment:
Note: This is NOT how I intend the final package to look like. This is more of a sample experiment.
But if you overall have ideas how it should look like, and how we can integrate directly with https://github.com/kubernetes/kubernetes/pull/71589,  you may comment here :)

Reply:
possibly the file name is WIP, but given the file also has decoding it might make sense to call it something else?

regarding #71589, i guess if we wrap the universal decoder like so and make potential consumers of the component config to use these functions instead of the core ones, they will get the strict verifications as part of the tool chain.
SGMT and we are already doing that in kubeadm.

if the UniversalStrictDecoder is not making it in api machinery maybe that's what we have to do.